packages/react-server-cli/src/commands/start.js

```diff
@@ -238,15 +242,41 @@ export default function start(options){
 		httpsOptions,
 		longTermCaching,
 		customMiddlewarePath,
+		compileOnStartup,
 	} = options;
 
-	const {serverRoutes, compiler} = compileClient(options);
+	const routesFileLocation = path.join(process.cwd(), '__clientTemp/routes_server.js');
+	let serverRoutes,
+		compiler;
 
 	const startServers = () => {
 		// if jsUrl is set, we need to run the compiler, but we don't want to start a JS
 		// server.
 		let startJsServer = startDummyJsServer;
 
+		if ((hot === false || hot === "false") && (compileOnStartup === false || compileOnStartup === "false")) {
+			serverRoutes = new Promise((resolve, reject) => {
+				fs.access(routesFileLocation, fs.constants.R_OK, (err) => {
+					if (err) {
+						reject("You must manually compile your application when compileOnStartup is set to false.");
+					} else {
+						// We need to replace the promise returned by the compiler with an already-resolved promise with the path
+						// of the compiled routes file.
+						resolve(routesFileLocation);
+					}
+				});
+			});
+
+			// mock the compiler object used by the various JS servers so that the compiler.run function always succeeds.
+			// This will allow the JS servers to work properly, thinking that the compiler actually ran.
+			compiler = {};
+			compiler.run = (cb) => {
+				cb(null, null);
+			};
+		} else {
+			({serverRoutes, compiler} = compileClient(options));
```

Comment:
Me too.  Here's the issue I have: I defined `serverRoutes` and `compiler` previously using `let`.  ES6 destructuring only works like this:
```
let {a, b} = stuff;
const {c, d} = stuff;
let e, f;
({e, f} = stuff);
```
This is a syntax error: `{a, b} = stuff;`.  I could put a comment above that line so that it's clear?